src/3d/qgspointcloudlayer3drenderer.cpp

```diff
@@ -110,12 +114,28 @@ void QgsPointCloudLayer3DRenderer::readXml( const QDomElement &elem, const QgsRe
   mLayerRef = QgsMapLayerRef( elem.attribute( QStringLiteral( "layer" ) ) );
 
   QDomElement elemSymbol = elem.firstChildElement( QStringLiteral( "symbol" ) );
-  if ( !mSymbol )
-    mSymbol = qgis::make_unique< QgsPointCloud3DSymbol >();
+  int renderingStyleInt = elemSymbol.attribute( QStringLiteral( "rendering-style" ), QStringLiteral( "0" ) ).toInt();
+  QgsPointCloud3DSymbol::RenderingStyle renderingStyle = static_cast<QgsPointCloud3DSymbol::RenderingStyle>( renderingStyleInt );
```

Comment:
Instead of an enum/int storage here, it would be better to use a string "id" value. This would make it slightly more future proof if we ever allowed the possibility of custom 3rd party renderers.

Reply:
I thought you'll tell me that we will need translation if I use strings directly

Reply:
agreed with Nyall - better to use strings. But not the translatable strings from GUI :-) Best to use e.g. `single-color`, `color-ramp`, `rgb` - and have two utility functions to convert between enum values and strings.

Reply:
I think I just confused storing the rendering style with having rendering style enum or not.